packages/model-registry/upstream/frontend/src/odh/utils/deploymentUtils.ts

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import { useDeploymentsState } from '../hooks/useDeploymentsState';
+import { KnownLabels } from '../k8sTypes';
+import { ModelVersion } from '../../app/types';
+
+export type DeploymentDetectionResult = {
+  hasDeployment: boolean;
+  loaded: boolean;
+};
+
+export const useModelDeploymentDetection = () => {
+  const { deployments, loaded } = useDeploymentsState();
+  const hasModelVersionDeployment = React.useCallback((mvId: string): DeploymentDetectionResult => {
+    if (!loaded || !deployments) {
+      // Conservative approach: assume deployment exists during loading to prevent flicker
+      // This keeps the archive button disabled until we're sure there are no deployments
+      return { hasDeployment: true, loaded };
+    }
+
+    const hasDeployment = deployments.some(deployment => {
+      const isInferenceService = deployment.model?.kind === 'InferenceService';
+      const modelVersionId = deployment.model?.metadata?.labels?.[KnownLabels.MODEL_VERSION_ID];
+      return isInferenceService && modelVersionId === mvId;
+    });
```

Comment:
Hmm, don't we also need to check that it matches the `KnownLabels.REGISTERED_MODEL_ID`?

Also, I'm a little confused about why this filtering is necessary. Don't we already have the filtering we need achieved by the labelSelectors we passed into MRDeploymentsContextProvider, which is what affects the deployments we get from `useDeploymentsState`? Maybe I'm missing something there.